### PR TITLE
feat :Connect to api

### PR DESCRIPTION
**Pull request by taher-software**
In this branch, I've connected my application to an API in order to make add and remove in remote server. Thus I've added the following changes :100: 
- Configure thunk middleware.
- Create action creator that returns a function to perform conditional dispatch.
- Change components to allow them interact with new action creators .

Please review my changes and give me your feedback before to move forward the next stage. 
Looking forward your feedback.
Best,

### CODE REVIEW

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @Taher-web-dev,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that has no descriptive README documentation.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,25 +1,112 @@
...
+});
+
+const manageBookFailed = (error) => ({
+  type: MANAGE_BOOK_FAILURE,
+  payload: {
+    error,
+  },
+});
+
+const addBookToApi = (data) => {
```
Comment (Mihndim2020) [inline src/redux/books/books.js:36]: - [ ] Please make sure  `add book` and `remove book` features should persist all changes in the server, refreshing the page after adding books, the books no longer displays, please kindly fix that.

File: src/components/Books/BooksList.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 import React from 'react';
 import { useSelector, useDispatch } from 'react-redux';
-import { removeBook } from '../../redux/books/books';
+import { removeExistBook } from '../../redux/books/books';
 
 const BooksList = () => {
-  const books = useSelector((state) => state.booksReducer);
+  const books = useSelector((state) => state.booksReducer.books);
```
Comment (Mihndim2020) [inline src/components/Books/BooksList.js:6]: - [ ] The books do not load when you refresh the page, please make sure a server requests is made in the `componentDidMount lifecycle` method or with `react hooks`.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @Taher-web-dev 🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**